lib/controller.js

```diff
@@ -641,6 +663,213 @@ controller.tesselFirmwareVerion = function(opts) {
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    }
+
+    rtm.once('select', function(label, index) {
+      rtm.close();
+      controller.menu.clear();
+
+      // Identify the Exit command by first letter (all other entries start with numbers/index)
+      if (label[0] !== 'E') {
+        callback(tessels[index].name, index, resolve, reject);
+        // controller.ssh.runSSH(index, opts, tessels, resolve, reject);
+      } else {
+        // going to clear screen and calling exit to resolve promise
+        controller.menu.clear();
+        controller.menu.exit(resolve);
+      }
+    });
+    //  });
+  },
+  makeMenu: function(tessel, index, rtm) {
+    if (tessel.lanConnection) {
+      if (!tessel.lanConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (authorized) \n');
+      }
+
+    } else if (tessel.usbConnection) {
+      if (!tessel.usbConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (authorized) \n');
+      }
+    }
+    // FIXME: Workaround for trouble with dynamic added menu elements
+    rtm._draw();
+  },
+  showMenu: function(rtm, resolve, reject) {
+    // Menu navigation
+    process.stdin.pipe(rtm.createStream()).pipe(process.stdout);
+
+    // raw mode for gain ability to navigate up and down within the menu
+    process.stdin.setRawMode(true);
+
+    // cb 
+    rtm.on('close', function() {
+      process.stdin.setRawMode(false);
+      process.stdin.end();
+    });
+    rtm.on('error', function(e) {
+      process.stdin.setRawMode(false);
+      reject(e);
+    });
+  },
+  seek: function(opts) {
+    // to be able to replace the seeker in testing mode
+    // return Tessel.seekTessels(opts);
+    return Tessel.get(opts);
+  },
+  clear: function() {
+    // selected exit!
+    process.stdout.write('\u001b[2J\u001b[0;0H');
+  },
+  exit: function(resolve) {
+    resolve();
+    //process.exit();
+  }
+};
+controller.ssh = {
+  runSSH: function(id, opts, tessels, resolve, reject) {
+    // clear console
+    debug('runSSH...\n root@' + tessels[id].lanConnection.host);
+    if (opts.menu) {
+      // because the test needs to resolve the promise ...
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    } else {
+      // if this is no test, starting the menu as child process
+      controller.menu.clear();
+      var ch = require('child_process')
+        .spawn('ssh', ['-i',
+          opts.path,
+          'root@' + tessels[id].lanConnection.ip
+        ], {
+          stdio: 'inherit'
+        });
+      logs.info('Connect to ' + tessels[id].lanConnection.host + '...');
+
+      // FIXME: There is no handler for poweroff Tessel ! Needs to be fixed within the firmware...
```

Comment:
I think it is not possible to fix this problem. The terminal freeze if you power down the Tessel 2 virtual machine. If you do `poweroff` logged in to a Tessel 2 by `t2 root` (ssh) this behavior is bad ... but there is no easy solution. I would expect it is fixing poweroff command by looking for open terminals and exit them with error code before power down. The comment is due to opening an issue or make a central note about knowledge about this behavior.